test/api/fake_api.dart

```diff
@@ -171,12 +176,16 @@ class FakeApiConnection extends ApiConnection {
     Uri? realmUrl,
     int? zulipFeatureLevel = eg.futureZulipFeatureLevel,
     Account? account,
+    Map<String, String> Function()? userAgentHeader,
```

Comment:
It looks like this is always the same expression, when not null: `testBinding.userAgentHeader`.

How about making it a boolean instead? Like `useBinding: true`. That'd reduce the amount of noise at the call sites.

Then I think it can actually be a boolean up at the base class, too.

After that change, I think `buildUserAgent` can move to `ApiConnection` (e.g. as a private static), which is where it most logically belongs. We'd lose memoization between different `ApiConnection` instances, but that's fine — we aren't constantly making new connection instances.